File: powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        result = 1
        if n < 0:
            n = -n
            return 1/self.myPow(x,n)
        elif not n:
            return 1
        elif n%2:
            return x * self.myPow(x, n-1)
        
        return self.myPow(x*x,n/2)

File: powx-n.cpp
#define ll double

class Solution {
public:
    double myPow(double x, int n) {
        ll tempn = n;
        n = abs(n);
        ll base = x;
        ll ans = 1;
         while (n > 0){
             if(n & 1) ans *= base;
             base *= base;

             n >>= 1;
         }

         if(tempn < 0) return 1/ans;
        return ans;
    }
};